### PR TITLE
Release 3.4.0

### PR DESCRIPTION
**Pull request by chenliangde**
- Made it possible to override category by hooking to nosto events
- Add a switch to disable sending customer information to nosto
- Send an empty cart to nosto if the cart is empty to empty the cart in nosto server

### CODE REVIEW

File: classes/helpers/NostoHelperConfig.php [hunk 1]
```diff
@@ -371,6 +372,32 @@ public static function saveCartUpdateEnabled($enabled)
         return self::saveSetting(self::CART_UPDATE_SWITCH, $enabled);
     }
 
+    /**
+     * Is customer tagging and order buyer tagging feature enabled
+     * @return bool true if customer tagging and order buyer tagging has been enabled, false otherwise
+     */
+    public static function isCustomerTaggingEnabled()
+    {
+        return (bool)Configuration::get(
```
Comment (mridang) [inline classes/helpers/NostoHelperConfig.php]: Why aren't you using `read`?

File: classes/models/NostoCategory.php [hunk 2]
```diff
@@ -39,7 +39,7 @@ public static function loadId($idCategory)
 
     public function __construct($category)
     {
-        parent::__construct($category, 'nosto_category');
+        parent::__construct($category);
```
Comment (supercid) [inline classes/models/NostoCategory.php]: I think if the child constructor is the same as the parent, we can remove it

File: classes/models/NostoSearch.php [hunk 3]
```diff
@@ -23,13 +23,13 @@
...
-use \Nosto\Object\MarkupableString;
+use \Nosto\Object\SearchTerm as NostoSDKSearchTerm;
 
-class NostoSearch extends MarkupableString
+class NostoSearch extends NostoSDKSearchTerm
 {
     public function __construct($searchTerm)
     {
-        parent::__construct($searchTerm, 'nosto_search_term');
+        parent::__construct($searchTerm);
```
Comment (supercid) [inline classes/models/NostoSearch.php]: I think if the child constructor is the same as the parent, we can remove it

File: classes/models/NostoCategory.php (hunk 2 above)
Comment (chenliangde) [reply classes/models/NostoCategory.php]: Fixed

**Pull request merged**